plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/TrinoIcebergRestCatalogFactory.java

```diff
@@ -108,6 +113,18 @@ public synchronized TrinoCatalog create(ConnectorIdentity identity)
             icebergCatalog = icebergCatalogInstance;
         }
 
-        return new TrinoRestCatalog(icebergCatalog, catalogName, sessionType, trinoVersion, typeManager, uniqueTableLocation);
+        ImmutableMap<String, String> credentials = securityProperties.get().entrySet().stream()
+                // OAuth2Properties.SCOPE is not set as scope passed through credentials is unused in https://github.com/apache/iceberg/blob/229d8f6fcd109e6c8943ea7cbb41dab746c6d0ed/core/src/main/java/org/apache/iceberg/rest/auth/OAuth2Util.java#L714-L721
+                .filter(property -> ImmutableSet.of(TOKEN, CREDENTIAL).contains(property.getKey()))
+                .collect(toImmutableMap(Map.Entry::getKey, Map.Entry::getValue));
```

Comment:
How about using Maps.filterKeys method in Guava? 
```java
// OAuth2Properties.SCOPE is not set as scope passed through credentials is unused in
// https://github.com/apache/iceberg/blob/229d8f6fcd109e6c8943ea7cbb41dab746c6d0ed/core/src/main/java/org/apache/iceberg/rest/auth/OAuth2Util.java#L714-L721
Map<String, String> credentials = Maps.filterKeys(securityProperties.get(), key -> Set.of(TOKEN, CREDENTIAL).contains(key));
```

Set.of is fine because this is a local usage.

Reply:
yea more cleaner. Thanks